src/Uno.UI/UI/Xaml/IFrameworkElementImplementation.macOS.tt

```diff
@@ -69,17 +70,18 @@ namespace <#= mixin.NamespaceName #>
 			try
 			{
 				var newWindow = Window;
-				var superView = Superview;
 
 				if(_currentWindow != newWindow)
 				{
 					if(newWindow != null)
 					{
-						if(_superViewRef?.GetTarget() == null && superView != null)
+						var parent = GetDependencyObjectAncestor(Superview) as NSView;
+						parent ??= Superview;
+						if(_superViewRef?.GetTarget() == null && parent != null)
 						{
-							_superViewRef = new WeakReference<NSView>(superView);
-							SyncBinder(superView, newWindow);
-							((IDependencyObjectStoreProvider)this).Store.Parent = superView;
+							_superViewRef = new WeakReference<NSView>(parent);
```

Comment:
Still not sure about what should be set here, the "real" Superview or the ancestor that we may have found that is a `DependencyObject`?

Reply:
I'm not sure either - it only seems to be checked here, and TBH I'm not clear on what the intent of the check is.

Any thoughts @jeromelaban ?

Reply:
But even for the `_superViewRef`? We shouldn't set that to the actual Superview regardless of it being a DependencyObject?

Reply:
@kazo0 I would leave it as you have it, setting it to `parent` (the actual dependency object), but I'm still not sure what the check is even for.